aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -291,6 +293,24 @@ static void threshold_kernel(TensorIterator& iter, Scalar threshold, Scalar valu
   });
 }
 
+namespace {
+
+template <typename T>
+void GeluCUDAKernelImplInternal(const Tensor& X, Tensor* Y) {
```

Comment:
can you avoid passing Tensors as `Tensor *`?  It's not standard C++ PyTorch code (is there any other example in the codebase that does this?).  Depending on the use case, you can use `const Tensor &`, `Tensor &` or `Tensor`.

Reply:
I just want to make things clear that pass by pointer means it is a output variable or it will be changed in the function. Just for readability reason. I can change it to Tensor&.

https://google.github.io/styleguide/cppguide.html#Output_Parameters

Reply:
See: https://github.com/pytorch/pytorch/wiki/Writing-Python-in-cpp-(a-manifesto)

Also, if you look at any `_out` function (translates to python with `out=` parameter), we use `Tensor&` already.  Although note that this kind of bogus, because reassigning the reference is almost never correct, but you should just follow the convention for now.

But the right way to think about this is you are already passing a (smart) pointer.  Passing a pointer to a smart pointer is almost never what you want.  And as noted in the link, const is essentially meaningless here, so trying to use static types for readability doesn't really work either (unless you implement ConstTensor).